vscode/src/utilities/logger.ts

```diff
@@ -0,0 +1,49 @@
+import * as winston from "winston";
+import * as vscode from "vscode";
+import * as path from "path";
+import { ExtensionPaths } from "../paths";
+import { OutputChannelTransport } from "winston-transport-vscode";
+import { getConfigLogLevel } from "./configuration";
+import { KONVEYOR_OUTPUT_CHANNEL_NAME } from "@editor-extensions/shared";
+
+// Create logger instance
+export function createLogger(paths: ExtensionPaths): winston.Logger {
+  // Create or get the output channel
+  const outputChannel = vscode.window.createOutputChannel(KONVEYOR_OUTPUT_CHANNEL_NAME, "log");
+
+  // Create log file path
+  const logFile = path.join(paths.serverLogs.fsPath, "konveyor-extension.log");
+
+  const logger = winston.createLogger({
+    level: getConfigLogLevel(),
+    format: winston.format.combine(
+      winston.format.timestamp(),
+      winston.format.errors({ stack: true }),
+      winston.format.json(),
+    ),
+    transports: [
+      // File transport
+      new winston.transports.File({
+        filename: logFile,
+        maxsize: 10 * 1024 * 1024, // 10MB
+        maxFiles: 3,
+      }),
+      // VS Code output channel transport
+      new OutputChannelTransport({
```

Comment:
Can we set a different level for vscode transport? Thinking we keep the output limited to WARN / ERROR

Reply:
Set the level in the vscode output channel to warn AND the console to "silly" for development.